libs/libc/unistd/lib_getopt_common.c

```diff
@@ -393,6 +393,14 @@ int getopt_common(int argc, FAR char * const argv[],
 
           /* Check for the end of the argument list */
 
+          if (go->go_optind >= argc)
+            {
+              /* We don't need to check arguments anymore, we are finished */
+
+              go->go_binitialized = false;
+              return ERROR;
+            }
+
           go->go_optptr = argv[go->go_optind];
```

Comment:
```suggestion
          go->go_optptr = go->go_optind < argc ? argv[go->go_optind] : NULL;
```

Reply:
Just as alternative shorter solution to get the same result

Reply:
Is ternary operator recommended if it can be shorter or simple in NuttX project ?
I sent PR to here for the first time.

Reply:
I'm not sure how to answer your question. In general ternary operator is widely used in NuttX code. Also I think that code duplication should be avoided and if we can achieve this with ternary operator then it should be used.